taking-turns.js

```diff
@@ -0,0 +1,209 @@
+    readlineInterface.question(questionText, resolve);
+  });
+}
+// end readline setup
+
+
+/****  FUNCTIONS  ****/
+
+// function holding the game logic for the computer guessing
+async function computerGuesses() {
+  // variables/constants
+  let MAX = process.argv[2] || 100;    // get max number from argv
+  let MIN = 0;
+  let currHigh = MAX;
+  let currLow = MIN;
+  let guessCount = 0;
+  let currGuess;
+  let highOrLow; // holds the answer of higher or lower
+  let yesOrNo; // holds the answer of if the guess is right or wrong
+  
+  
+  // console.log introductory message
+  console.log(`Please think of a number between ${MIN} and ${MAX} (inclusive).
+  I will try to guess it.`);
+  
+  // get a random number for the computer's guess with Math.random()
+  //currGuess = randomInteger(currLow, currHigh);
+  currGuess = findMid(currLow, currHigh);
+  guessCount++; //increment guessCount
+  
+  // ask if the computer guess is the player's number
+  yesOrNo = await ask(`Is it... ${currGuess}? (y or n) `);
+  
+  // validate user response
+  yesOrNo = await validateInput(yesOrNo, currGuess);
+
+  // need to loop through if statement so long as yesOrNo == 'N'
+  while (yesOrNo.toUpperCase() == "N") {
+    // if player says No (N) guess is not right, ask if it is higher or lower
+    highOrLow = await ask(`Is it higher (H), or lower (L)? `);
+    
+    while (highOrLow.toUpperCase() !== "H" && highOrLow.toUpperCase() !== "L") {
+      console.log(`I don't understand what you're telling me!`);
+      highOrLow = await ask(`Is it higher (H), or lower (L)? `);
+    }
+    
+    // update currLow or currHigh based on highOrLow
+    if (highOrLow.toUpperCase() == "H") {
+      if (currGuess == currHigh) {
+        console.log(`It can't be higher than ${currGuess}, earlier you said it was lower than ${currHigh+1}!`);
+        guessCount--;   // this adjusts for bad response
+      } else {
+        currLow = currGuess + 1; // since it is higher, add 1
+      }
+    // else highOrLow == "L"
+    } else {
+      if (currGuess == currLow) {
+        console.log(`It can't be lower than ${currGuess} if you said it was higher than ${currLow-1}`);
+        guessCount--;   // this adjusts for bad user response
+      } else {
+        currHigh = currGuess - 1; // since lower, subtract 1
+      }
+    }
+    
+    // recalculate guess, increment guessCount
+    // currGuess = randomInteger(currLow, currHigh);
+    currGuess = findMid(currLow, currHigh);
+
+    guessCount++;
+    
+    // ask player about currGuess
+    yesOrNo = await validateInput(await ask(`Is it... ${currGuess}? (y or n) `), currGuess);
+    
+  }
+
+  console.log(`Your number was ${currGuess}!`);
+  console.log(`I guessed it in ${guessCount} times!`);
+    
+  // exit the game
+  //process.exit();
+}
+
+
+// function containing the game logic for the user guessing
+async function userGuesses() {
+  // variables & constants
+  let MAX = process.argv[2] || 100;    // get max number from argv
```

Comment:
Naming conventions in JavaScript are to use lowercase letters for variable names.  Capitalized names are reserved for class names (which we haven't taught you about yet)